pkg/services/hcloud/server/server.go

```diff
@@ -342,23 +420,29 @@ func (s *Service) reconcileLoadBalancerAttachment(ctx context.Context, server *h
 	return reconcile.Result{}, nil
 }
 
-func (s *Service) createServer(ctx context.Context) (*hcloud.Server, error) {
-	// get userData
+func (s *Service) createServerFromImageName(ctx context.Context) (*hcloud.Server, *hcloud.Image, error) {
 	userData, err := s.scope.GetRawBootstrapData(ctx)
 	if err != nil {
 		record.Warnf(
 			s.scope.HCloudMachine,
 			"FailedGetBootstrapData",
 			err.Error(),
 		)
-		return nil, fmt.Errorf("failed to get raw bootstrap data: %s", err)
+		return nil, nil, fmt.Errorf("failed to get raw bootstrap data: %s", err)
 	}
-
-	image, err := s.getServerImage(ctx)
+	image, err := getServerImage(ctx, s.scope.HCloudClient, s.scope.HCloudMachine, s.scope.HCloudMachine.Spec.Type, s.scope.HCloudMachine.Spec.ImageName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("createServerFromImageName: failed to get server image: %w", err)
```

Comment:
we have another wrapper one level up, so you don't have to add the function name here IMO